src/context/AuthContext.tsx

```diff
@@ -0,0 +1,36 @@
+import { createContext, ReactNode, useEffect, useState } from 'react'
+import { GetToken } from '../components/auth/security'
+
+interface IAuthContext {
+  isAuth: boolean
+  login: (token: string) => void
+  logout: () => void
+}
+
+export const AuthContext = createContext({} as IAuthContext)
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const token = GetToken()
```

Comment:
You should validate if this token is valid in back end too

Reply:
https://github.com/samueldurantes/xau/blob/master/packages/web/src/contexts/auth.tsx#L20 using something like that?

Reply:
Yep, in the backend

Reply:
Provide me more context. When and where should I validate it? When I'm logging in I generate a new token and should validate it right away?